include/swift/AST/DiagnosticsSema.def

```diff
@@ -3749,8 +3749,13 @@ NOTE(add_self_to_type,none,
 WARNING(warn_unqualified_access,none,
         "use of %0 treated as a reference to %1 in %2 %3",
         (Identifier, DescriptiveDeclKind, DescriptiveDeclKind, DeclName))
+WARNING(self_refers_to_method,none,
+       "'self' refers to the method '%0.self', which may be unexpected",
+       (StringRef))
```

Comment:
You can use `Type` here.

Reply:
I was originally using Type, but then the error would look like 
`"'self' refers to the method '%'TypeWithSelfMethod'.self', which may be unexpected",`
instead of:
`"'self' refers to the method '%TypeWithSelfMethod.self', which may be unexpected",`

Not sure how to stop those extra quotes without using StringRef?